store/sqlstore/role_supplier.go

```diff
@@ -24,12 +24,22 @@ type Role struct {
 }
 
 func FromRole(role *model.Role) *Role {
+	permissionsMap := make(map[string]bool)
+	permissions := ""
+
+	for _, permission := range role.Permissions {
```

Comment:
It solves the problem of repeated permissions, but generate a corner case when I try to modify the role, and give a repeated permission, generating as result, the same permissions string, and, in the end, throwing a 500 error.

Reply:
I've been thinking of adding `CreateAt` `UpdateAt` `DeleteAt` fields to the role schema in anticipation of "custom roles" in phase 2 or 3. I guess that would also solve this issue of the error 500 on no changes, as that's what the other store functions do (e.g. `updateChannel`).

Reply:
I agree, It's something needed anyway, and fix the problem :)

Reply:
Is the `DeleteAt` needed here? I don't think so. Anyway if we going to store deleted roles, we'll need to remove the Unique constrain in the name, I think.

Reply:
Yes, DeleteAt is needed when we have custom roles, and we still need the unique constraint, because otherwise when deleting a role we'd have to go through and remove it from every User/TeamMember/ChannelMember it's assigned to.

Reply:
Unless we move over to proper join tables, I guess.

Reply:
Ok, sorry, I was concerned about the workflow (a user delete a role, and create a new role with the same Name) I understand that it will be something like (you create another one with the same "DisplayName", and the underneath name will be generated by us).

Reply:
Will make another update tomorrow with those fields added.